src/ol/control/ScaleLine.js

```diff
@@ -274,8 +306,125 @@ class ScaleLine extends Control {
     }
 
   }
-}
 
+  /**
+   * @private
+   * @param {number} width The current width of the scalebar.
+   * @param {number} scale The current scale.
+   * @param {string} suffix The suffix to append to the scale text.
+   * @returns {string} The stringified HTML of the scalebar.
+   */
+  createScaleBar(width, scale, suffix) {
+    let mapScale = Math.round(this.getScaleForResolution());
+    mapScale = '1 : ' + mapScale.toLocaleString().replace(/,/g, '.');
```

Comment:
I don't think we want to always replace `,` with `.` here.

For example, for my current locale:
```js
> (1e3 + 1e-2).toLocaleString().replace(/,/g, '.')
1.000.01
```

Instead, these two lines could be:
```js
const mapScale = '1 : ' + Math.round(this.getScaleForResolution()).toLocaleString();
```